### PR TITLE
docs: Adding GenAI content

### PR DESCRIPTION
**Pull request by franciscojavierarceo**
**Checklist:**
- [ ] You have [signed off your commits](https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits)
- [ ] Ensure you follow best practices from our guide. [Contributing](https://github.com/kubeflow/website/blob/master/content/en/docs/about/contributing.md). 
- [] You have included screenshots when changing the website style or adding a new page.

**Description of your changes:**

### Issue

Closes: #

### Labels

---

### CODE REVIEW

File: content/en/_index.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
 +++
 
 <!-- Hero section -->
-{{<blocks/cover title="Kubeflow" subtitle="The Machine Learning Toolkit for Kubernetes" image_anchor="center" color="dark">}}
+{{<blocks/cover title="Kubeflow" subtitle="The AI Platform for Kubernetes" image_anchor="center" color="dark">}}
```
Comment (thesuperzapper) [inline content/en/_index.html]: I think we need to carefully consider any significant rebranding "The Machine Learning Toolkit for Kubernetes" has been the tagline for many years.

Additionally, the word "toolkit" is much more appropriate for Kubeflow as an open source project. We are trying to be the standard set of tools for people to create platforms from, rather than being a platform ourselves.

If anything, we should consider calling ourselves "The Artificial Intelligence Toolkit for Kubernets".

File: content/en/_index.html [hunk 2]
```diff
@@ -14,6 +14,10 @@
       Contribute
       <i class="fas fa-pencil-alt ml-1"></i>
     </a>
+    <a class="btn btn-lg btn-info mx-1 mb-3 mb-sm-0 px-3 px-md-4 py-2" href="/docs/genai/">
+      GenAI 
+      <i class="fas fa-brain ml-1"></i>
+    </a>
```
Comment (thesuperzapper) [inline content/en/_index.html]: It's important to not have too many options for the "intended action" when a user comes to a homepage.

I suggest if we want people to know about GenAI, we have a section under the Introduction/Getting Started which is already linked in its own button, and leave the two buttons as "Get Started" and "Contribute".

File: content/en/_index.html [hunk 3]
```diff
@@ -29,7 +33,7 @@ <h3 class="section-head">
       <span class="border-bottom pb-2">What is Kubeflow?</span>
     </h3>
     <p class="mx-auto col-11 col-xl-7 px-0">
-      Kubeflow makes artificial intelligence and machine learning simple, portable, and scalable.
+      Kubeflow is a platform that makes artificial intelligence and machine learning simple, portable, and scalable.
```
Comment (thesuperzapper) [inline content/en/_index.html]: Same as my other comment, we are not a platform, but a collection of tools which can form a platform.

Also, I think the proposed text is less punchy from a marketing perspective.

We are trying to sell the "why" not the "what" in this sentence.

**Review by thesuperzapper — DISMISSED** [review]
We should discuss rebranding of homepage in a dedicated PR, as this will cause a LOT of discussion and likely block the other changes for a long time.

File: content/en/_index.html (hunk 1 above)
Comment (franciscojavierarceo) [reply content/en/_index.html]: We had a conversation with @castrojo from CNCF and I believe the perception of Kubeflow is that it is a platform and we should just lean into how users perceive us and clarify that we are a collection of projects/components/tools.

Also just because it was the tagline for many years doesn't mean we can't update it based on what's more appropriate today. :)

File: content/en/_index.html (hunk 2 above)
Comment (franciscojavierarceo) [reply content/en/_index.html]: I disagree here. I think Kubeflow has lost mind-share in the GenAI era. We received explicit feedback from the outgoing KSC that we need to make Kubeflow more visible in the GenAI conversation. This PR was my first attempt to do that.

File: content/en/_index.html (hunk 3 above)
Comment (franciscojavierarceo) [reply content/en/_index.html]: >a collection of tools which can form a platform

To me, that's equivalent. Most people using Kubeflow consider it a platform and we should just lean into that.

File: content/en/_index.html (hunk 2 above)
Comment (thesuperzapper) [reply content/en/_index.html]: My comment was about how to effectively use buttons on the homepage, not about GenAI.

Having three buttons is just bad UX, and will lead to confusion about what the intended action of the user is.

File: content/en/_index.html (hunk 1 above)
Comment (thesuperzapper) [reply content/en/_index.html]: Sure, but as with any significant rebranding (changing the tagline), we need to work through all the options so we get the best final result.

Off the top of my head, there are many things to discuss, like:

1. should even shorten "AI" or use "Artificial Intelligence"
2. are we using the word "Platform" or "Toolkit", or something else
3. should we include any reference to "ops".
4. how long should it be (its a tagline, so probably only a few words)
5. anything else people might want to discuss for a significant rebrand

File: content/en/_index.html (hunk 3 above)
Comment (thesuperzapper) [reply content/en/_index.html]: This sentence went through many iterations recently, and I am not sure we need to re-litigate it.

I am happy with the current state:

> Kubeflow makes artificial intelligence and machine learning simple, portable, and scalable. We are an ecosystem of [Kubernetes](https://kubernetes.io/) based components for each stage in [the AI/ML Lifecycle](https://www.kubeflow.org/docs/started/architecture/#kubeflow-components-in-the-ml-lifecycle) with support for best-in-class open source [tools and frameworks](https://www.kubeflow.org/docs/started/architecture/#kubeflow-ecosystem).

File: content/en/_index.html (hunk 2 above)
Comment (franciscojavierarceo) [reply content/en/_index.html]: I disagree it's bad UX.

Buttons are a mechanism to (1) succinctly give users information and (2) redirect them to more information. 

Usually people prioritize real estate above the fold on whatever is most important to the top of funnel and my guess is that based on the current state of AI, GenAI (and Kubeflow's ability to support those needs) is a good fit for it.

CC @varodrig @chasecadet @StefanoFioravanzo  thoughts?

**Comment by thesuperzapper** [comment]
To avoid accidentally merging an update to the homepage, lets put a hold.

If other content in the PR is important to merge sooner, I would suggest making a separate PR for the homepage updates and rebranding 

@kubeflow/kubeflow-steering-committee

/hold

**Comment by franciscojavierarceo** [comment]
Sure I've created the other PR here: https://github.com/kubeflow/website/pull/4125

File: content/en/_index.html (hunk 1 above)
Comment (andreyvelich) [reply content/en/_index.html]: @franciscojavierarceo Please can we only add the GenAI content in this PR and update the index page in the followup PR ?

File: content/en/docs/genai/_index.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
++++
+title = "GenAI"
```
Comment (andreyvelich) [inline content/en/docs/genai/_index.md]: Shall we call this page GenAI with Kubeflow ?

File: content/en/docs/genai/use-cases.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
++++
+title =  "GenAI Use Cases"
+description = "GenAI Use Cases Powered by Kubeflow"
+weight = 10
+aliases = ["/genai/use-cases/"]
++++
+
+# Powering GenAI Use Cases with Kubeflow
+
+Kubeflow is an open-source MLOps platform that unifies key components—Kubeflow Pipelines for workflow orchestration, the Training Operator for distributed model training, Katib for automated tuning, Feast for feature management, and KServe for scalable serving—to power every stage of a Generative AI (GenAI) application lifecycle. From generating synthetic data to retrieval-augmented generation (RAG), fine-tuning large language models (LLMs), hyperparameter optimization, inference at scale, and evaluation, Kubeflow’s modular, Kubernetes-native architecture makes building end-to-end GenAI pipelines both reproducible and production-ready.
```
Comment (andreyvelich) [inline content/en/docs/genai/use-cases.md]: I am not sure if we should re-iterate what is Kubeflow here, we can just cross-reference users to this page: https://deploy-preview-4123--competent-brattain-de2d6d.netlify.app/docs/started/introduction/#what-is-kubeflow.

We can just say here that
```
Kubeflow Projects are powered every stage of GenAI application lifecycle.
From generating synthetic data to retrieval-augmented generation (RAG), fine-tuning large language models (LLMs), hyperparameter optimization, inference at scale, and evaluation,
Kubeflow’s modular, Kubernetes-native architecture makes building end-to-end GenAI pipelines both reproducible and production-ready.
```

File: content/en/_index.html (hunk 1 above)
Comment (franciscojavierarceo) [reply content/en/_index.html]: Already did here https://github.com/kubeflow/website/pull/4125

File: content/en/docs/genai/use-cases.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+1. [Hyperparameter Optimization](#hyperparameter-optimization)  
+1. [Inference at Scale](#inference-at-scale)  
+1. [Model Evaluation](#model-evaluation)  
+1. [Beyond Core Use Cases](#beyond-core-use-cases)  
+
+---
+
+## Synthetic Data Generation
+
+When real data is scarce or sensitive, Kubeflow Pipelines automates the creation of high-fidelity synthetic datasets using techniques like GANs, VAEs, and statistical copulas. By defining parameterized pipeline components that:
```
Comment (andreyvelich) [inline content/en/docs/genai/use-cases.md]: Would be nice to have some reference links here and don't use acronyms (e.g. GANs, VAEs)

File: content/en/docs/genai/use-cases.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+3. **Tuning Retrieval Parameters**  
+   - Use Katib to sweep over `top_k`, `temperature`, etc., and optimize metrics like BLEU  
+
+This workflow is detailed in the [Katib RAG blog post](https://blog.kubeflow.org/katib/rag/) and the [Feast + Docling RAG tutorial](https://docs.feast.dev/tutorials/rag-with-docling).
+
+--- 
+
+## Scaling RAG Data Transformation with Spark
+
+Preprocessing massive document collections for RAG pipelines—text cleaning, chunking, and embedding generation—can become a bottleneck. Kubeflow integrates with the Spark Operator to run distributed Spark jobs on Kubernetes, allowing you to:
```
Comment (andreyvelich) [inline content/en/docs/genai/use-cases.md]: Can we have a link to the Spark Operator docs ?

File: content/en/docs/genai/use-cases.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+- Text Chunking: Normalize, tokenize, and split content into fixed-size passages with overlap.
+- Distributed Embedding: Call embedding services (e.g., OpenAI, HuggingFace) inside Spark UDFs or map operations to parallelize across executor pods.
+- Direct Write to Feature Store: Persist chunk embeddings and metadata back into Feast’s offline store or vector store without intermediate bottlenecks.
+
+
+---
+
+## Fine-Tuning LLMs
+
+Domain-specific fine-tuning of pre-trained LLMs is streamlined by the Kubeflow Training Operator’s legacy Trainer API:
```
Comment (andreyvelich) [inline content/en/docs/genai/use-cases.md]: Let's link this document here for now: https://www.kubeflow.org/docs/components/trainer/legacy-v1/user-guides/fine-tuning/

Once we implement the torchtune Trainer, we can reference the docs (cc @Electronic-Waste)

File: content/en/docs/genai/use-cases.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+For details, see the [Kubeflow Trainer fine-tuning guide](https://www.kubeflow.org/docs/components/trainer/legacy-v1/user-guides/fine-tuning/).
+
+## Hyperparameter Optimization
+Automated tuning is essential to maximize model performance:
+
+Katib Experiments let you define hyperparameter search spaces and optimization objectives (e.g., minimize loss, maximize BLEU).
+
+GSoC-2024 Hyperparameter Optimization API provides programmatic Experiment/Trial management, metric collection, and result tracking.
+
+Explore the GSoC 2024 project here: [GSoC-2024 Project 4: HPO API](https://blog.kubeflow.org/gsoc-2024-project-4/).
```
Comment (andreyvelich) [inline content/en/docs/genai/use-cases.md]:
```suggestion
Katib Experiments let you define hyperparameter search spaces and optimization objectives (e.g., minimize loss, maximize BLEU).

Katib allows you to effortlessly [optimize hyperparameters of LLMs using distributed PyTorchJobs](https://blog.kubeflow.org/gsoc-2024-project-4/).

```

cc @helenxie-bit

File: content/en/docs/genai/use-cases.md [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+
+## Inference at Scale
+After training and tuning, KServe delivers scalable, framework-agnostic inference:
```
Comment (andreyvelich) [inline content/en/docs/genai/use-cases.md]: Would be nice to have link to KServe docs
cc @terrytangyuan

File: content/en/docs/genai/use-cases.md [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+
+
+## Model Evaluation
+Rigorous evaluation guards against drift and degradation:
+
+- Pipeline-Embedded Eval Steps
+    - Statistical benchmarks for synthetic data
+    - Text metrics (BLEU, ROUGE) for generation quality
+
+- Early-Stopping & Tuning
```
Comment (andreyvelich) [inline content/en/docs/genai/use-cases.md]: Do you want to have link to EarlyStopping docs ?
https://deploy-preview-4123--competent-brattain-de2d6d.netlify.app/docs/components/katib/user-guides/early-stopping/

File: content/en/docs/genai/use-cases.md [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+
+- Notebooks UI for interactive development (Jupyter, VSCode)
+
+- Model Registry for artifact versioning & lineage
+
+- Spark & Batch Operators for data-intensive preprocessing
+
+- Feast [supports vector similarity search](https://docs.feast.dev/tutorials/rag-with-docling) for RAG and other next-gen AI workloads
+
+- Each component plugs seamlessly into Kubernetes, ensuring portability and consistency.
```
Comment (andreyvelich) [inline content/en/docs/genai/use-cases.md]:
```suggestion
- Each Kubeflow Project plugs seamlessly into Kubernetes, ensuring portability and consistency.
```

File: content/en/_index.html (hunk 1 above)
Comment (andreyvelich) [reply content/en/_index.html]: Oh, sorry.
I left a few comments to the GenAI page, we can address them in #4125

**Comment by juliusvonkohout** [comment]
/lgtm
from my side

**Comment by juliusvonkohout** [comment]
@franciscojavierarceo you need to /unhold to merge it "Waiting for status to be reported — Not mergeable. Should not have do-not-merge/hold label."
